llama.h

```diff
@@ -348,6 +357,25 @@ extern "C" {
                                  float * scores,
                                    int   capacity);
 
+    // Get a llama layer
+    LLAMA_API struct llama_layer * llama_get_layer_from_model(
+            struct llama_model * model,
+                           int   layer);
+    
+    LLAMA_API struct ggml_tensor * llama_get_model_tok_embeddings(struct llama_model * model);
+    LLAMA_API struct ggml_tensor * llama_get_model_norm          (struct llama_model * model);
+    LLAMA_API struct ggml_tensor * llama_get_model_output        (struct llama_model * model);
+
+    LLAMA_API struct ggml_tensor * llama_get_layer_attention_norm(struct llama_layer * layer);
+    LLAMA_API struct ggml_tensor * llama_get_layer_wq            (struct llama_layer * layer);
+    LLAMA_API struct ggml_tensor * llama_get_layer_wk            (struct llama_layer * layer);
+    LLAMA_API struct ggml_tensor * llama_get_layer_wv            (struct llama_layer * layer);
+    LLAMA_API struct ggml_tensor * llama_get_layer_wo            (struct llama_layer * layer);
+    LLAMA_API struct ggml_tensor * llama_get_layer_ffn_norm      (struct llama_layer * layer);
+    LLAMA_API struct ggml_tensor * llama_get_layer_w1            (struct llama_layer * layer);
+    LLAMA_API struct ggml_tensor * llama_get_layer_w2            (struct llama_layer * layer);
+    LLAMA_API struct ggml_tensor * llama_get_layer_w3            (struct llama_layer * layer);
+
```

Comment:
This does not scale well with adding support for new non-LLaMA models in the near future.
Would it be possible to get the tensors by name?

```c
LLAMA_API struct ggml_tensor * llama_get_model_tensor(const char * name);
```

Reply:
Certainly, replaced it by:

```c
LLAMA_API struct ggml_tensor * llama_get_model_tensor(struct llama_model * model, const char * name);
```